platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java

```diff
@@ -115,23 +122,26 @@
 @DisableCachingByDefault(because = "Not worth caching")
 public abstract class CreateStartScripts extends ConventionTask {
 
-    private File outputDir;
-    private String executableDir = "bin";
     private final Property<String> mainModule;
     private final Property<String> mainClass;
-    private Iterable<String> defaultJvmOpts = new LinkedList<>();
-    private String applicationName;
-    private String optsEnvironmentVar;
-    private String exitEnvironmentVar;
-    private FileCollection classpath;
+    private final Property<String> optsEnvironmentVar;
+    private final Property<String> exitEnvironmentVar;
+    private final Property<String> executableDir;
     private final ModularitySpec modularity;
     private ScriptGenerator unixStartScriptGenerator = new UnixStartScriptGenerator();
     private ScriptGenerator windowsStartScriptGenerator = new WindowsStartScriptGenerator();
+    private final DirectoryProperty outputDir;
+    private final Property<String> applicationName;
```

Comment:
❓ Can we implement these via abstract getters instead?

Reply:
Yeah, we can do that